patroni/postgresql/mpp/citus.py

```diff
@@ -389,16 +389,11 @@ def bootstrap(self) -> None:
         if self._config['database'] != self._postgresql.database:
             conn = connect(**conn_kwargs)
             try:
-                database = self._config['database']
-                sql = """DO $$
-BEGIN
-    PERFORM * FROM pg_catalog.pg_database WHERE datname = {0};
-    IF NOT FOUND THEN
-        CREATE DATABASE {1};
-    END IF;
-END;$$""".format(quote_literal(database), quote_ident(database, conn))
                 with conn.cursor() as cur:
-                    cur.execute(sql.encode('utf-8'))
+                    cur.execute('CREATE DATABASE {0}'.format(
+                        quote_ident(self._config['database'], conn)).encode('utf-8'))
+            except Exception as e:
+                logger.warn('Exception when creating database: %r', e)
```

Comment:
The problem indeed exists, because I forgot that CREATE DATABASE is not transactional :(
But the fix is be wrong. We need to catch `DuplicateDatabase` exception and ignore it (or better have debug log in).

Reply:
Thanks for the comment, I will fix this.